### DISCUSSION TITLE
Troubles loading and merging Font Awesome 5

### DISCUSSION THREAD

**Discussion by Uki99** [ocornut/imgui#4499] [New Users Build/Link/Run issues ONLY!]
Hello!

Im rather new to ImGui and i am just playing around with basic settings of GUI itself.
I am well aware you have examples of how to do exactly this at [FONTS.md](https://github.com/ocornut/imgui/blob/master/docs/FONTS.md), but i still somehow cannot load and merge Font Awesome!

I successfully loaded Segoeui as my default display font, and Font Awesome seems to load and everything compiles and everything works normally except my icon shows up as question mark. I also tried loading font awesome from memory using `io.Fonts->AddFontFromMemoryCompressedBase85TTF()` and converting **fa-regular-400.ttf** to base85 encoded C style byte array. That also yield the very same result, and no errors! Why is that?

This is my "do once" setup in infinite loop:
```cpp
if (!Data::InitImGui)
{
    ImGui::CreateContext();
    ImGuiIO& io = ImGui::GetIO();
    io.ConfigFlags = ImGuiConfigFlags_NoMouseCursorChange;
    io.Fonts->AddFontFromFileTTF("C:\\Windows\\Fonts\\segoeui.ttf", 20.f);

    ImFontConfig config;
    config.MergeMode = true;
    static const ImWchar icon_ranges[] = { ICON_MIN_FA, ICON_MAX_FA, 0 };
                
     // Yields same result as loading normal .ttf file
    //io.Fonts->AddFontFromMemoryCompressedBase85TTF(fa_regular_400_base85, 16.0f, &icons_config, icons_ranges);

    io.Fonts->AddFontFromFileTTF("C:\\Users\\mypc\\Desktop\\fa-regular-400.ttf", 13.0f, &config, icon_ranges);
    io.Fonts->Build();

    ImGui_ImplWin32_Init(Data::hWindow);
    ImGui_ImplDX9_Init(pDevice);
    Data::InitImGui = true;
}
```

And drawing it like this:
```cpp
	ImGui_ImplDX9_NewFrame();
	ImGui_ImplWin32_NewFrame();
	ImGui::NewFrame();

	if (Data::ShowMenu)
	{
		Theme();      // Applies custom styles
		RenderGUI();  // GUI itself
		GUILogic();   // Logic
	}

	ImGui::EndFrame();
	ImGui::Render();
	ImGui_ImplDX9_RenderDrawData(ImGui::GetDrawData());
```

Where RenderGUI() is:
```cpp
void RenderGUI(void)
{
    ImGui::Begin("My window", &Base::Data::ShowMenu, ImGuiWindowFlags_NoCollapse | ImGuiWindowFlags_NoResize);

    // Menu here
    {
        ImGui::Button(ICON_FA_SEARCH " Search");
    }

    ImGui::End();
}
```

**Comment by ocornut** [comment]
Hello!
Hmm at first glance the code seems right.
You can use the Metrics window to inspect your loaded/merge fonts, see if the FontAwesome glyphs are listed there?
If they are listed, perhaps the encoding type used in the `ICON_FA_SEARCH` macro is not compatible with your compiler setting? There's a raw C version of those headers which should work everywhere.

**Reply by Uki99** [reply]
How very weird. I tried with ICON_FA_STAR, and it works! I tried with some symbols around it and it works, but for most it doesn't work.
This looks like problem with icon_ranges, but i can't really figure out why because i did not change anything...

**Comment by Uki99** [comment]
Apparently, the fonts are loaded by ImGui just as expected. I suspect there might be an issue with headers, like @ocornut mentioned.
More specifically, with [IconsFontAwesome5.h](https://github.com/juliettef/IconFontCppHeaders/blob/main/IconsFontAwesome5.h)
I switched to using IconsFontAwesome4.h and regular fontawesome-webfont.ttf and everything works just fine! I do wonder if the header was really the issue or some weird setting in my compiler... But the format of those macros is the same across all those headers.

**Reply by ocornut** [reply]
Have you digged in Metrics>Fonts to see what glyphs are available?
Try to look at the value of ICON_FA_SEARCH and see how it compares to the values appearing there.

**Reply by Uki99** [reply]
I'm not quite sure about all the functionality of fonts in ImGui, but i've dug around some functions and i got this: [screenshot](https://prnt.sc/1qwbtme)
The font seems to be loaded, but i am not sure about that return value of the function "FindGlyph()", the macro for ICON_FA_SEARCH is "\xef\x80\x82" and i am not even sure if i am using this function correctly.
ShowMetrics window doesn't have anything regarding fonts and ShowFontSelector() let's me select only the first font i have loaded.
FontAwesome4 works nicely though.
But notice how that star is displayed correctly and ICON_FA_SEARCH as question mark. I've absolutely no idea...
